tests/rustdoc/jump-to-def-intra-doc.rs

```diff
@@ -0,0 +1,21 @@
+// This test ensures that the same link is generated in both intra-doc links
```

Comment:
Please put this file in the existing `tests/rustdoc/jump-to-def` dir.

Reply:
Do you mean "there's no intra-doc link linking to Debug and PartialEq *in core*"? Because line 10 and 11 literally test that a link exists.  Or are you not counting those as intra-doc because they aren't written as a doc comment?  If that's the case, you should probably remove the comment on line 9 that describes them as doc comments.

Reply:
Arf, I'm so bad at explaining.

So:

```rust
//! [PartialEq] [Debug]

#[derive(Debug, PartialEq)]
pub struct Foo;
```

Before this PR, if the `//! [PartialEq] [Debug]` doc comment was not present, in "jump to def", the links for `Debug` and `PartialEq` would be wrongly generated there (in the source code pages). Because it wasn't in the cache and so the path computation was wrong.